### ISSUE TITLE
Typedefs and `ptr_arg` lint

### ISSUE THREAD

**Issue by stanislav-tkach** [rust-lang/rust-clippy#1981]
I have the following situation:
```rust
type Value = Vec<u8>;

fn foo(_: &Value) {}
```
Typedef is used for clarity, but Clippy [produces](https://play.rust-lang.org/?gist=7f94e3fa12b35627ec98aedb56babbc8&version=stable) a [warning](https://github.com/rust-lang-nursery/rust-clippy/wiki#ptr_arg):
> warning: writing `&Vec<_>` instead of `&[_]` involves one more reference and cannot be used with non-Vec-based slices. Consider changing the type to `&[...]`

Probably this lint can be disabled for typedes?

**Comment by oli-obk** [comment]
> Probably this lint can be disabled for typedefs?

hmm... I think it would be ok to disable the lint on typedefs if the function is not public. But if it's public, you're gonna limit your callers needlessly.

**Comment by DarkEld3r** [comment]
In my case both function and typedef isn't public.

**Comment by algesten** [comment]
My types are public but it's an internal library where I clarify what is what using type defs. However I won't ever need `&str` instead of `&String`. I'm not sure about this lint :/

**Comment by oli-obk** [comment]
Maybe we should rewrite this lint as a `PreExpansionPass` and simply check for literal `&String` and `&Vec<T>` even if this causes false positives when someone creates their own types named `String` or `Vec`

**Comment by algesten** [comment]
That sounds like a great idea!
